### DISCUSSION TITLE
melonDs Enchanted Update ! ( Concept & Issues )

### DISCUSSION THREAD

**Discussion by Bittu2002** [rafaelvcaetano/melonDS-android#645] [Ideas]
Hey @rafaelvcaetano ! I am Back Again To Help You In melonDs Development. **This time,** I give you Some Suggestion as well as some improvements ideas with Images. I cant know coding So I only help you by giving ideas  :(

**Ok ! Lets Start Concept & Issues**

**1. Icon Fixes**

- In Menu
This look more good ! Please Try It.
![icon](https://user-images.githubusercontent.com/90685841/134765203-08e49bd3-b3c2-4291-a7b6-99530d8f87a1.jpg)

- Main App New Logo
![fmbete5kvoy31](https://user-images.githubusercontent.com/90685841/134707862-2b14ec8a-b35e-46e3-98cf-6d6e66b3b3b2.jpg)

**2. Quick save & Quick Load Button**
![quick save load](https://user-images.githubusercontent.com/90685841/134701583-7c87cc8f-14eb-4c5e-a0c1-36ee271e1a31.jpg)

**3. Take ScreenShot In Pause Menu ( Both Screens in one .png file &  pixel perfect )**
![Screenshot_2021-09-24-18-30-05-703_me magnum melonds](https://user-images.githubusercontent.com/90685841/134708881-30633ef5-0db1-4fa5-9e2c-208a62002d45.jpg)

**4. Extra Useful Option in Rom Option**
- Add A Create Shortcut ( I know this feature is already there but in some Launchers ( Like my miui System launcher ) It is not working
- A Rom Info option to see information about rom like
• Company name 
• Rom icon
• Rom name
• Rom size
• Which region
• Language
• Locate directory ( clicking this we go to direct to the rom file in file explore )
• Etc. Lot more :D
![rom info](https://user-images.githubusercontent.com/90685841/134710516-1556329e-3c19-46de-bc8a-d16ad091807f.jpg)

**5. Improve Theme !**

- New Amoled Theme ( Black Theme For Amoled Screen )
![Screenshot_2021-09-24-18-50-48-550_me magnum melonds](https://user-images.githubusercontent.com/90685841/134711339-30163987-fa38-47f5-9972-8449ffd18da0.jpg)

- Fix Theme
![Screenshot_2021-09-24-22-23-39-843_me magnum melonds](https://user-images.githubusercontent.com/90685841/134715114-1ff16ee3-09b4-47fe-9a24-1adaedfd958f.jpg)

**6. 0.5x Speed (Slow-Mo)**
- For Those who Want To See Nds 3D Games (Mario Kart :P) Running In Slow Motion (60 fps)
![Screenshot_2021-09-24-18-50-24-088_me magnum melonds](https://user-images.githubusercontent.com/90685841/134712176-a0b78ed7-16c3-4c7e-a588-d72f217a4b59.jpg)

**7. Category Menu**
Right Now All Settings are in One place So User have to find a Specific Option Take Time !
By Adding Category Menu All Options Are Easy To find
Like => First User Open Setting Then If He/She Wanted to access General settings So They Click General option after that only general Settings Will Appear ! 
![Screenshot_2021-09-24-22-23-39-843_me magnum melonds (1)](https://user-images.githubusercontent.com/90685841/134718014-c2a843f9-b0fe-4018-8b22-7585c0dd93b0.jpg)

**8. New Info Page**
• ChangeLog
• Help
• About

**9. Google Dive Backup Options ( toggle On & Off)**
- Upload Save Files
- Upload Screenshot ( only which taken by app )
- Upload Emulator Settings ( like key mapping, layout etc.)
- Upload Cheat Data Base

**10. By Default Icon Filtering  / Video Filter Set To None (Pixelated)**
Why Not ! Which user Need Game Blurred By Default  :D

***Thanks For Making This Emulator For Free And Ad Free :D
Love You Bro !***

**Comment by ashgx** [comment]
emmmmm I hope the simulator can add translation

**Comment by griffi-gh** [comment]
I prefer the current app icon

**Comment by albion1993** [comment]
> **@rafaelvcaetano Its Been 3 Days Waiting For Your Response :(**
> Ignoring me? Or Not Want To Reply ? Or Not Want To Adds These Feature ? Or Busy I Think You are not busy Because You are replying to Others...I Research , Make Images , And write this ! Its Take Lots Of Time. Just I Thinking You Get Happy By Knowing Someone is there To Give You Good Feedback To Help In The Development Of This MelonDs Port.
> Sorry I Disturb You. Sorry If My Words Hurts. I Dont Do This From Next Time. But Please Reply This Last Time :)

LoL! Just leave the dev to port the app like he likes to do it. Don't insist  that much for something only because you think it's the best, nobody asked you to do that anyways.

**Comment by rafaelvcaetano** [comment]
Come on guys. No need to be mean. I've been a bit busy and I've answered other users because their questions are much simpler. I've read the list of suggestions and been thinking about them so that I can give constructive feedback. I'll try to give an in-depth answer later today.

And Bittu is just trying to improve the app by giving his opinion and there's absolutely nothing wrong with that. Suggestions are more than welcome.

**Comment by griffi-gh** [comment]
everything except the new icon is good

**Comment by Bittu2002** [comment]
> Come on guys. No need to be mean. I've been a bit busy and I've answered other users because their questions are much simpler. I've read the list of suggestions and been thinking about them so that I can give constructive feedback. I'll try to give an in-depth answer later today.
> 
> And Bittu is just trying to improve the app by giving his opinion and there's absolutely nothing wrong with that. Suggestions are more than welcome.

I Take My Words Back ! ^_^ ( Now Happy )
Sorry Dear Developer, I thought You Just Ignored My Ideas That Why I Am Just Go In Negative a Little Bit : ) But Not Anymore Now I Am Happy ! Please Take Your Time And Reply When ever You Have Time. Now I know That You Readed My Ideas That What I want. And From Next Time If You Read My Idea Just React It With Thumps Up ( Emoji ) So I Know That You Readed Ok ! Always With You Bro.

**Comment by RayyanAnsari** [comment]
> A Rom Info option to see information about rom like
• Company name
• Rom icon
• Rom name
• Rom size
• Which region
• Language
• Locate directory ( clicking this we go to direct to the rom file in file explore )
• Etc. Lot more :D

This shouldn't be too hard to implement, I already added structs for a ROM Info dialog upstream.
https://github.com/Arisotura/melonDS/blob/master/src/NDS_Header.h
(also has extra DSi fields that Arisotura added.) 
![image](https://user-images.githubusercontent.com/68647953/134982834-195b906a-444e-4e97-90be-0692191f2204.png)

**Comment by rafaelvcaetano** [comment]
OK. Let's review these suggestions:

> 1. Icon Fixes


Not really keen on changing the icon. The upper suggestion is not match melonDS's design and the bottom one simply looks more bland.

Regarding the menu changes, they go against the Material Design guidelines. The current 3 dot icon opens a pop-up with multiple options, not just the settings, and that's exactly what users expect when clicking on such an icon. Not for a different screen to be opened.

I'll have to reject this suggestion.

> 2. Quick save & Quick Load Button


Buttons to quickly save and load a save state are a good idea! Although they will probably have a different design, I'll add this suggestion to my TODO list.

> 3. Take ScreenShot In Pause Menu ( Both Screens in one .png file & pixel perfect )


Already added to my TODO list as per your last suggestions. By the way, PNG files are lossless, meaning that they retain the original image quality.

> 4. Extra Useful Option in Rom Option


* **Create shortcut**: Has already been suggested by another user and it's already in my TODO list
* **ROM Info**: While this is usually useful for PC versions, Android users are a bit more casual and this information is not really important for them. Anyway, I'll add this to my TODO list since, as Rayyan shown, is already available in the PC version, but with a super low priority since it won't add much value.

> 5. Improve Theme!


An Amoled theme is interesting and I'll consider it. But the changes to the already existing themes do not convince me. The current colors were extracted from the icon to try to keep a consistent design, but I known that, the accent color in particular, is not the best looking. I'll review them to see if I find a better mix while keeping a consistent design.

> 6. 0.5x Speed (Slow-Mo)


This would be better suited for a separate button so that both options (slow down and speed up) are available at the same time. But I don't see this option being really useful. I'll add this to the TODO list with low priority.

> 7. Category Menu


I've also been thinking about this for some time. The list of settings has grown a lot since the start and it may be time to group everything in sub-screens. I'll add this to my TODO list.

> 8. New Info Page


This is bound to come eventually, which should include legal stuff like open source licenses and privacy policy.

> 9. Google Dive Backup Options ( toggle On & Off)

As stated previously, this was already suggested by another user and it's in my TODO list. Backing up cheat databases won't be done since that would take a lot of space and take a lot of time to synchronize.

> 10. By Default Icon Filtering / Video Filter Set To None (Pixelated)


Not sure. Maybe...? I also prefer no filters. But every other emulator I've used seems to come with the linear filter enabled by default, leading me to believe that it's what most users use. And since I don't want to add analytics to collect user information, I'll never know... I'll leave it as is for know but consider it for the future.

_____________

I'll also move this to a discussion and delete the existing one to keep the issue list cleaner.
